src/simplnx/Utilities/ThreadedAlgorithm.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include "simplnx/Filter/IFilter.hpp"
+#include "simplnx/simplnx_export.hpp"
+
+#include <chrono>
+#include <cstdint>
+
+namespace nx::core
+{
+
+class SIMPLNX_EXPORT ThreadedAlgorithm
+{
+
+public:
+  ThreadedAlgorithm(const IFilter::MessageHandler& mesgHandler, const std::atomic_bool& shouldCancel);
+  virtual ~ThreadedAlgorithm() noexcept;
+
+  ThreadedAlgorithm(const ThreadedAlgorithm&) = delete;
+  ThreadedAlgorithm(ThreadedAlgorithm&&) noexcept = delete;
+  ThreadedAlgorithm& operator=(const ThreadedAlgorithm&) = delete;
+  ThreadedAlgorithm& operator=(ThreadedAlgorithm&&) noexcept = delete;
+
+  const std::atomic_bool& getCancel() const;
```

Comment:
```suggestion
  bool shouldCancel() const;
```
A thread-safe fetch function